src/plugins/share/public/components/url_panel_content.tsx

```diff
@@ -77,6 +86,10 @@ export class UrlPanelContent extends Component<Props, State> {
     this.state = {
       exportUrlAs: ExportUrlAsType.EXPORT_URL_AS_SNAPSHOT,
       useShortUrl: false,
+      showTopNavMenu: false,
```

Comment:
Since this component is reused across different apps, these will be applied everywhere. Can we make a HOC component around `UrlPanelContent` that will be dashboard specific?

Reply:
I agree with @majagrubic here - the `share` plugin should be generic and be usable in all applications, so application-specific logic should be part of the application.

I propose the following API:
Currently `toggleShareContextMenu` is called in https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx#L858 (eventually defined in https://github.com/elastic/kibana/blob/master/src/plugins/share/public/services/share_menu_manager.tsx#L59 )

Let's add a new optional property to `ShowShareMenuOptions` called
```ts
// Props are all props of the UrlPanelContent component
urlParamExtensions: Array<{
  paramName: string;
  component: ComponentType<Props & { setParamValue: (value: string) => void; }>
}>
```

This should allow us to move all of the dashboard-specific stuff into the dashboard controller and keep the generic logic of rendering a list of components, keeping state for a list of params and appending them to the url here in the share plugin.

Reply:
IMO These options are as relevant to Visualize as well, don't you think? @flash1293

Reply:
I thought the plan for this would be it becoming a single integration point for sharing, also usable from apps like Canvas, Graph and so on. But maybe that's not necessary, we can also put it behind a prop which defaults to `false` directly in the share code and only disable it for dashboards for now. No strong opinion on this, whatever you are more comfortable with :)